typings/styled-components.d.ts

```diff
@@ -53,9 +53,7 @@ type ThemedStyledComponentFactories<T> = {
 export interface ThemedBaseStyledInterface<T> extends ThemedStyledComponentFactories<T> {
   <P, TTag extends keyof JSX.IntrinsicElements>(tag: TTag): ThemedStyledFunction<P, T, P & JSX.IntrinsicElements[TTag]>;
   <P, O>(component: StyledComponentClass<P, T, O>): ThemedStyledFunction<P, T, O>;
-  <P extends { theme: T; }>(component: React.ComponentClass<P>): ThemedStyledFunction<P, T, WithOptionalTheme<P, T>>;
-  <P>(component: React.ComponentClass<P>): ThemedStyledFunction<P, T>;
-  <P extends { [prop: string]: any; theme?: T; }>(component: React.StatelessComponent<P>): ThemedStyledFunction<P, T, WithOptionalTheme<P, T>>;
+  <P extends { [prop: string]: any; theme?: T }>(component: React.ComponentType<P>): ThemedStyledFunction<P, T, WithOptionalTheme<P, T>>;
```

Comment:
Have you tried and tested that theme type `T` (type of `theme` prop) is actually respected by the compiler? I afraid having ` [prop: string]: any ` would eat all other props, including `theme`.

Reply:
There are lots of existing tests for this; see [here](https://github.com/styled-components/styled-components/tree/e4f59b047cb92c693ed2218dca934eef703fa2e7/typings/tests/themed-tests). The `[prop: string]: any` trick (which you can see was already being used previously in one of the overloads) is necessary to ensure that a `theme` prop of the wrong type is not provided.

Reply:
Yes, makes sense. Just wanted to double-check.